frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -135,6 +137,47 @@ frappe.views.QueryReport = Class.extend({
 			frappe.render_grid({grid:this.grid, title:__(this.report_name)});
 		}
 	},
+	pdf_report: function() {
+		if(this.html_format) {
+			var content = frappe.render(this.html_format,
+				{data: this.dataView.getItems(), filters:this.get_values(), report:this});
+
+			//Render Report in HTML
+			var html = frappe.render_template("print_template", {content:content, title:__(this.report_name)});
+		} else {
+			var columns = this.grid.getColumns();
+			var data = this.grid.getData().getItems();
+			var content = frappe.render_template("print_grid", {columns:columns, data:data, title:__(this.report_name)})
+
+			//Render Report in HTML
+			var html = frappe.render_template("print_template", {content:content, title:__(this.report_name)});
+		}
+
+		//Create a form to place the HTML content 
+		var formData = new FormData();
+
+		//Push the HTML content into an element
+		formData.append("html", html);
+		var blob = new Blob([], { type: "text/xml"});
+		//formData.append("webmasterfile", blob);
+		formData.append("blob", blob);
```

Comment:
Good question, I did try it without the blob but the API would not process the form data via POST if without. I'm not sure why, perhaps it was needed to explicitly define the type i.e. text/xml. Either way I would have ideally wanted to just a simple AJAX call but the APIs doesnt seem to like accepting params if the AJAX is POST. Tried troubleshooting down all the way till response.py and I couldn't get further than that. I suspect it may have to be due to a very low level python response behavior (could be wrong though).